assets/app/scripts/directives/nav.js

```diff
@@ -20,86 +20,78 @@ angular.module('openshiftConsole')
       templateUrl: "views/_sidebar-main-nav-item.html"
     };
   })
-  .directive('projectNav', function($timeout, $location, $filter, LabelFilter) {
+  .directive('projectNav', function($timeout, $location, $filter, LabelFilter, DataService) {
     return {
-      restrict: 'E',
+      restrict: 'EA',
```

Comment:
I'd rather be consistent right now, if we decide to change it down the road
then we would change them all then.

We save attribute directives to mean something different.  Attribute
directives add some kind of functionality.  This is a best practice from
the angular docs:

"*When should I use an attribute versus an element?* Use an element when
you are creating a component that is in control of the template. The common
case for this is when you are creating a Domain-Specific Language for parts
of your template. Use an attribute when you are decorating an existing
element with new functionality."


On Mon, Aug 31, 2015 at 9:38 AM, Ben Petersen <notifications@github.com>
wrote:

> In assets/app/scripts/directives/nav.js
> <https://github.com/openshift/origin/pull/4419#discussion_r38312029>:
>
> >      return {
> > -      restrict: 'E',
> > +      restrict: 'EA',
>
> Well, if we ever decided we wanted valid html, we could then put our
> directives on <div>'s as data-* attributes & accomplish the same thing.
> I'm guessing we aren't going to worry about that, but figured could leave
> the option open.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4419/files#r38312029>.
>